### PR TITLE
n3-4 Add security configuration based on Basic authentication.

### PR DESCRIPTION
**Pull request by dmkofgl**

### CODE REVIEW

File: src/main/java/dl/news/portal/web/controller/UserController.java [hunk 1]
```diff
@@ -13,14 +13,15 @@
...
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.*;
 
 import javax.persistence.EntityNotFoundException;
 
 @RestController
-@RequestMapping("/users")
+@RequestMapping(value = "/users",produces =MediaType.APPLICATION_JSON_VALUE )
```
Comment (avinahradau) [inline src/main/java/dl/news/portal/web/controller/UserController.java]: refactor formatting here

Comment (dmkofgl) [reply src/main/java/dl/news/portal/web/controller/UserController.java]: Done

File: src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @Override
+    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
+
+        UserDetails userDetails = userService.findUserByUsername(s)
+                .map(this::mapUser)
+                .orElseThrow(() -> new UsernameNotFoundException("No user found with username " + s));
+        return userDetails;
+    }
+
+    private UserDetails mapUser(User user) {
```
Comment (avinahradau) [inline src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java]: unlear name map User with what?

File: src/main/java/dl/news/portal/web/config/security/BasicAuthenticationConfig.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Bean
+    public UserDetailsService newsPortalUserDetailsService() {
+        return new UserDetailService();
+    }
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder(5);
```
Comment (avinahradau) [inline src/main/java/dl/news/portal/web/config/security/BasicAuthenticationConfig.java]: What does this magic digit mean?

**Review by avinahradau — CHANGES_REQUESTED** [review]

File: src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java (hunk 2 above)
Comment (dmkofgl) [reply src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java]: Map to UserDetails

File: src/main/java/dl/news/portal/web/config/security/BasicAuthenticationConfig.java (hunk 3 above)
Comment (dmkofgl) [reply src/main/java/dl/news/portal/web/config/security/BasicAuthenticationConfig.java]: It's bcript password encoder strength

**Review by avinahradau — APPROVED** [review]

**Pull request merged**